### PR TITLE
Task 5 - Import to S3

### PR DESCRIPTION
**Pull request by machulan**
[Task 5](https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task5-import-to-s3/task.md)

CloudFront:
https://d15zr8xzdj76yx.cloudfront.net

[import service API](https://neifoq9cqi.execute-api.eu-west-1.amazonaws.com/dev)
[get signed URL - GET /import](https://neifoq9cqi.execute-api.eu-west-1.amazonaws.com/dev/import)

FE PR:
https://github.com/nikinsta/nodejs-aws-fe/pull/4

All subtasks except unit tests are done.

### CODE REVIEW

**Review by maximsan — APPROVED** [review]
Everything except unit-tests - confirmed :)

File: import-service/handlers/importFileParser.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+              CopySource: resourcePath,
+              Key: newKey,
+            }).promise();
+            console.log(`importFileParser file is copied to ${newResourcePath}`);
+
+            await s3.deleteObject({
+              Bucket: BUCKET_PARAMS.name,
+              Key: key,
+            }).promise();
+            console.log(`File ${resourcePath} is removed`);
```
Comment (generalov) [inline import-service/handlers/importFileParser.ts:43]: I'd use the function `pipeline`:
```js
import * as util from 'util'; // node.js
import * as stream from 'stream'; // node.js
const pipeline = util.promisify(stream.pipeline);
////
await pipeline(s3ReadStream, csvParser().on('data', console.log));
const key = ...
```

File: import-service/handlers/importFileParser.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { APIGatewayProxyHandler, S3Event } from 'aws-lambda';
+import * as AWS from 'aws-sdk'
+import 'source-map-support/register';
+import * as csvParser from 'csv-parser';
+import { BUCKET_PARAMS } from '../utils/constants';
+import { makeErrorResponse, makeCreatedResponse } from '../utils/response';
+
+export const importFileParser: APIGatewayProxyHandler = async (event: S3Event, _context) => {
```
Comment (generalov) [inline import-service/handlers/importFileParser.ts:8]: `S3Handler` would be better. The handler should not return an HTTP response.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b9b6b0e9bd5de6ed18be0a3e07816f60fe6f2cfa/types/aws-lambda/trigger/s3.d.ts#L3

**Review by generalov — APPROVED** [review]
LGTM

**Pull request merged**